### DISCUSSION TITLE
Plugin support when running from Home Assistant

### DISCUSSION THREAD

**Discussion by gdombiak** [zwave-js/zwave-js-ui#3137] [General]
Hey everyone,

Thanks for creating this amazing and simple to use project! I have Home Assist running with zwave-js-ui integration on my RPi. Everything works very well! I then tried installing "@varet/zj2m-prom-exporter" plugin or my own developed plugin. The plugin that I created works very well when running zwave-js-ui standalone. Both plugins fail to load when running from inside HomeAssistant. Below you can see the error I'm getting.

> 2023-06-09 23:38:05.252 ERROR APP: Error while loading @varet/zj2m-prom-exporter plugin Cannot find module '@varet/zj2m-prom-exporter'
>Require stack:
>- /opt/server/app.js
>- /opt/server/bin/www.js
>Error: Cannot find module '@varet/zj2m-prom-exporter'
>Require stack:
>- /opt/server/app.js
>- /opt/server/bin/www.js
>at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
>at Function.Module._load (node:internal/modules/cjs/loader:920:27)
>at Module.require (node:internal/modules/cjs/loader:1141:19)
>at require (node:internal/modules/cjs/helpers:110:18)
>at startGateway (/opt/server/app.js:333:17)
>at /opt/server/app.js:847:9

Let me know if you need any other information to help troubleshoot this. I hope that running  zwave-js-ui plugins when running integrated with Home Assistant is supported. :)

Thanks,
Gaston

**Comment by kpine** [comment]
If you are talking about the **add-on**, then you'll have to make the request at https://github.com/hassio-addons/addon-zwave-js-ui. It's a separate project that packages ZUI in its own way. Is that what you are referring to by "integrated with Home Assistant"?

**Reply by gdombiak** [reply]
Thanks @kpine for the quick reply. You are correct, I'm referring to the addon. For now I will remove all zwave addons and just point the integration to my standalone zwavejsui server that works with plugins. I will also post in the proper forum to see if plugins are supported in the addon or this would be a new feature.

Thanks,
Gaston
